autosar/writer/behavior_writer.py

```diff
@@ -59,8 +59,6 @@ def writeInternalBehaviorXML(self,internalBehavior):
             for event in internalBehavior.events:
                 lines.extend(self.indent(self._writeEventXML(ws,event),2))
             lines.append(self.indent('</EVENTS>',1))
-        if len(internalBehavior.portAPIOptions)==0:

```

Comment:
This  would change behavior for all existing scripts expecting portAPIOptions to be created automatically for their applicationComponents. It would be much better if the internalBehavior class had a new boolean attribute called something like autoCreatePortAPIOptions which is automatically set to true for any ApplicationComponent but set to false for any NVBlockComponent (during their respective creation).
Then this line would look like this:
if (len(internalBehavior.portAPIOptions)==0) and (internalBehavior.autoCreatePortAPIOptions):

In package.py we should give a new option to the method _createInternalBehavior, like so:
def _createInternalBehavior(self, ws, swc, behaviorName, multipleInstance, autoCreatePortAPIOptions):

This new parameter can now be set to true from inside package.createApplicationSoftwareComponent but false when calling package.createNvBlockComponent. This way we can choose which component classes should automatically have autoCreatePortAPIOptions enabled by default. 
The InternalBehavior constructor itself should always have false on this option. Only if you use the package API should things be enabled by default.

I'm leaning towards rejecting this commit since the change I described above is an entirely different solution to the same problem but that doesn't break existing code. What do you think?

Reply:
Sounds like a better solution.

I will create a new request when I have the change in place.